### PR TITLE
Update Segment links

### PR DESCRIPTION
**Pull request by bryanlogan**
# Pull Request/Issue Resolution

#### Description of Change:
Updating links to point out that the SDK used is dependent on which Segment SDK is used.

Closes #SDK-3255

#### Is this change associated with a Braze feature/product release?
- [ ] Yes (**Insert Feature Release Date Here**)
- [ X ] No

---

<details>
<summary>✔️ Pull Request Checklist</summary>
<br>

- [ ] Check that you haven't removed any images (replacing an image with an updated one of the same name is fine), as this breaks the French site
- [ ] Check that all links work.
- [ ] Ensure you have completed [our Contributors License Agreement](https://www.braze.com/docs/cla/).
- [ ] Tag @Timothy-Kim and @KellieHawks as a reviewer when your work is **done and ready to be reviewed for merge**. Are you an internal product manager? Reference the internal reviewing chart to tag the appropriate reviewer.
- [ ] Tag others as reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `assets` > `js` > `broken_redirect_list.js`

</details>

<details>
<summary>⭐ Helpful Wiki Shortcuts</summary>
<br>

- [Writing Style Guide](https://docs.google.com/document/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub)
- [Image Style Guide](https://docs.google.com/document/d/e/2PACX-1vRJSkwcjmjrTfLDagZccLpOMMyh5NN5SXRZSjz12cRAHbX4OrUmhvCmYpf_p5YB-9r4_jSOQLkicQIH/pub)
- [Styling Test Page](https://www.braze.com/docs/home/styling_test_page/)

</details>

<details>
<summary>❗ ATTN: For PR Reviewers</summary>
<br>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Read our [Previewing Documentation page](https://github.com/braze-inc/braze-docs/wiki/Previewing-and-Testing-Documentation) to see how to check the deployment.
  - [ ] Preview all changes in the linked Vercel environment by clicking the preview link in the vercel-bot comment in your PR.
</details>

<details>
<summary>❗ ATTN: Internal Reviewing Chart </summary>
<br>
<b>Work at Braze and not sure who to tag for review?</b> <br>Before tagging @timothy-kim or @KellieHawks for a general review, reference the following chart to see if a specific product vertical/reviewer applies to your pull request.
<br><br>
<table>
<tr>
    <td><b>Reviewer</b></td>
    <td><b>Product Vertical</b></td>
  </tr>
  <tr>
    <td>@josh-mccrowell-braze</td>
    <td>Monolith Deployments<br>Quality Infrastructure<br>Platform Infrastructure<br>Datalake<br>SDKs</td>
  </tr>
  <tr>
    <td>@kelliehawks</td>
    <td>Currents<br>Internal Tools<br>Product Partnerships<br>SMS<br>Customer Lifecycle, Identity and Permissions</td>
  </tr>
  <tr>
    <td>@bre-fitzgerald</td>
    <td>Reporting<br>Intelligence<br>User Targeting<br>IAM<br>Channels<br>FIX</td>
  </tr>
  <tr>
    <td>@lydia-xie</td>
    <td>Ingestion<br>Core Objects<br>Core Messaging<br>Messaging and Automation<br>Email (Composition and Infrastructure)</td>
  </tr>
</table>
</details>

### CODE REVIEW

File: _docs/_partners/data_and_infrastructure_agility/customer_data_platform/segment/segment.md [hunk 1]
```diff
@@ -70,7 +70,17 @@ When using a device-mode connection, similar to integrating the Braze SDK native
 {% tab Android %}
 
 {% alert important %}
-The Braze (Classic) Destination is the correct destination to use for Mobile device-mode integrations.<br><br>The source code for the [Android device mode](https://github.com/Appboy/appboy-segment-android) integration is maintained by Braze and is updated regularly to reflect new Braze SDK releases.
+The source code for the Android device mode integration is maintained by Braze and is updated regularly to reflect new Braze SDK releases.
+
+The Braze SDK you use will depend on which Segment SDK you use
```
Comment (KellieHawks) [inline _docs/_partners/data_and_infrastructure_agility/customer_data_platform/segment/segment.md:75]:
```suggestion
<br>
The Braze SDK you use will depend on which Segment SDK you use:
```

File: _docs/_partners/data_and_infrastructure_agility/customer_data_platform/segment/segment.md [hunk 2]
```diff
@@ -85,7 +95,15 @@ The source code for the [Braze Web Mode (Actions) destination](https://github.co
 {% tab iOS %}
 
 {% alert important %}
-The Braze (Classic) Destination is the correct destination to use for Mobile device-mode integrations.<br><br>The source code for the [iOS device mode](https://github.com/Appboy/appboy-segment-ios) integration is maintained by Braze and is updated regularly to reflect new Braze SDK releases.
+The source code for the iOS device mode integration is maintained by Braze and is updated regularly to reflect new Braze SDK releases.
+
+The Braze SDK you use will depend on which Segment SDK you use
```
Comment (KellieHawks) [inline _docs/_partners/data_and_infrastructure_agility/customer_data_platform/segment/segment.md:100]:
```suggestion
<br>
The Braze SDK you use will depend on which Segment SDK you use:
```

**Comment by bryanlogan** [comment]
https://braze-docs-en-git-sdk-3255segmentupdates-braze.vercel.app/docs/partners/data_and_infrastructure_agility/customer_data_platform/segment/segment#side-by-side-sdk-integration

The table layout could probably use some different color. Not sure if we want to marked the older versions as "old" or the first ones as "preferred".

**Review by radixdev — APPROVED** [review]

**Comment by josh-mccrowell-braze** [comment]
> https://braze-docs-en-git-sdk-3255segmentupdates-braze.vercel.app/docs/partners/data_and_infrastructure_agility/customer_data_platform/segment/segment#side-by-side-sdk-integration
> 
> The table layout could probably use some different color. Not sure if we want to marked the older versions as "old" or the first ones as "preferred".

I've been using the term "Legacy," and like how it avoids time-based words like "old." I think "Preferred" or "Recommended" are good terms for the SDKs we _want_ people to use.

**Review by josh-mccrowell-braze — APPROVED** [review]

File: _docs/_partners/data_and_infrastructure_agility/customer_data_platform/segment/segment.md [hunk 3]
```diff
@@ -85,7 +96,16 @@ The source code for the [Braze Web Mode (Actions) destination](https://github.co
...
-The Braze (Classic) Destination is the correct destination to use for Mobile device-mode integrations.<br><br>The source code for the [iOS device mode](https://github.com/Appboy/appboy-segment-ios) integration is maintained by Braze and is updated regularly to reflect new Braze SDK releases.
+The source code for the iOS device mode integration is maintained by Braze and is updated regularly to reflect new Braze SDK releases.
+
+<br>
+The Braze SDK you use will depend on which Segment SDK you use:
+
+| Segment SDK | Braze SDK |
+| ----------- | --------- |
+| [Analytics-Swift](https://github.com/segmentio/analytics-swift) | [Braze Segment Swift](https://github.com/braze-inc/analytics-swift-braze) |
+| [Analytics-iOS](https://github.com/segmentio/analytics-ios) | [Braze Segment iOS](https://github.com/Appboy/appboy-segment-ios) |
```
Comment (chshapiro) [inline _docs/_partners/data_and_infrastructure_agility/customer_data_platform/segment/segment.md:107]: This hasn't been productionized yet (as far as I'm aware) so we might not want it in public docs at this point

**Review by josh-mccrowell-braze — APPROVED** [review]
LGTM

**Pull request merged**